docs/html/development/release-process.rst

```diff
@@ -80,20 +80,13 @@ Creating a new release
 ----------------------
 
 #. Checkout the current pip ``master`` branch.
-#. Ensure you have the latest ``wheel``, ``setuptools``, ``twine`` and ``nox`` packages installed.
-#. Generate a new ``AUTHORS.txt`` (``nox -s generate_authors``) and commit the
-   results.
-#. Bump the version in ``pip/__init__.py`` to the release version and commit
-   the results. Usually this involves dropping just the ``.devN`` suffix on the
-   version.
-#. Generate a new ``NEWS.rst`` (``nox -s generate_news``) and commit the
-   results.
-#. Create a tag at the current commit, of the form ``YY.N``
-   (``git tag YY.N``).
-#. Checkout the tag (``git checkout YY.N``).
-#. Create the distribution files (``python setup.py sdist bdist_wheel``).
-#. Upload the distribution files to PyPI using twine
-   (``twine upload dist/*``).
+#. Ensure you have the latest ``nox`` and ``twine`` installed.
+#. Prepare for release using ``nox -s prepare-release -- YY.N``.
+   This will update the relevant files and tag the correct commit.
+#. Build the release artifacts using ``nox -s build-release -- YY.N``.
+   This will checkout the tag, generate the distribution files to be
+   uploaded and checkout the master branch again.
+#. Upload the distribution files to PyPI using ``twine upload dist/*``.
 #. Push all of the changes including the tag.
```

Comment:
It might be the occasion to clear up the workflow due to the fact that we cannot push directly on master.
What I did for 19.3 was:
- creating a pull request after "Prepare for release using ``nox -s prepare-release -- YY.N``" (without pushing the tag)
- wait for the CI and merge the PR with a merge commit *no squash or rebase* to keep the tagged commit
- only after the PR was merged did I push the matching tag and uploaded the new version to PyPI.

Reply:
> It might be the occasion to clear up the workflow due to the fact that we cannot push directly on master.

Let's address that in a follow up -- I disable the branch protections (so that I can merge the PR, before CI passes, by checking a checkbox to say "YES I KNOW WHAT I'M DOING") and then re-enable protections in the same minute.

The bugfix process also needs clarification, so there's that.